src/movies.js

```diff
@@ -1,22 +1,61 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    const direcFind = moviesArray.map((find) => find.director)
+    return direcFind;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    let stevenFilms = moviesArray.filter((el) => el.director === "Steven Spielberg" && el.genre.includes("Drama"));
+    return stevenFilms.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    if(!moviesArray.length) return 0;
+
+        const avg = moviesArray.reduce((acc ,val) => {
+    if(!val.score) {return acc}
+        return acc + val.score}, 0);
+    const decimal = avg / moviesArray.length;
+    return Math.round(decimal * 100) / 100;
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
```

Comment:
In the previous iteration you already created a function that calculates the average, so you can use that function inside this new function, and in this way spare a couple of lines of code:

```
function dramaMoviesScore(moviesArray) {
  const avgDram = moviesArray.filter((drama) => drama.genre.includes("Drama"));

  return scoresAverage(avgDram);
}
```